opal/datatype/opal_convertor_internal.h

```diff
@@ -22,10 +23,255 @@
 
 BEGIN_C_DECLS
 
-typedef int32_t (*conversion_fct_t)( opal_convertor_t* pConvertor, uint32_t count,
-                                     const void* from, size_t from_len, ptrdiff_t from_extent,
-                                     void* to, size_t to_length, ptrdiff_t to_extent,
-                                     ptrdiff_t *advance );
+#define COPY_TO_VECTOR   1
+#define COPY_FROM_VECTOR 2
+typedef size_t (*conversion_fct_t)( opal_convertor_t* pConvertor, int mode,
```

Comment:
Pass `const dt_elem_desc_t*` instead of this large number of parameters.

Reply:
I like that idea a lot, much cleaner.  Thanks, will change.